### ISSUE TITLE
Support for fifo-v1

### ISSUE THREAD

**Issue by cmeissl** [Smithay/smithay#1581]
[fifo-v1](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/256) has been merged

I tried to evaluate how we could support that in `smithay`, especially with the current `blocker` architecture.

So, my current plan is to use the pending state to hold latest `fifo_barrier`, which will need to be `Clone`.
In a `pre commit hook` we would inspect the two bool `set_barrier` and `wait_barrier`. First we would inspect
`wait_barrier`, when set we would inspect the `fifo_barrier` in the `pending` state and when the barrier is not signaled
insert a blocker waiting on that barrier. After that we inspect `set_barrier` and when set we create a new `fifo_barrier`
overriding the barrier stored in the `pending` state.
When merged into the transaction the `fifo_barrier` will be `cloned` to eventually reaching the `current` state.
During latching, the `fifo_barrier` of the `current` state will be cleared, allowing to make any `transaction` waiting
on that `fifo_barrier` to make progress.

One reason we need to use the `pending` state this way is that we can not use the `current` state as this could still be
blocked by some other constraint like `DmaBufBlocker`.

This should allow us to create a forward chain of `fifo_barrier`.
